src/cpp/server/server.cc

```diff
@@ -303,12 +323,21 @@ bool Server::Start() {
   return true;
 }
 
-void Server::Shutdown() {
+void Server::ShutdownInternal(gpr_timespec deadline) {
   grpc::unique_lock<grpc::mutex> lock(mu_);
   if (started_ && !shutdown_) {
     shutdown_ = true;
     grpc_server_shutdown_and_notify(server_, cq_.cq(), new ShutdownRequest());
     cq_.Shutdown();
+    SyncRequest* request;
+    bool ok;
+    while (SyncRequest::AsyncWait(&cq_, &request, &ok, deadline)) {
```

Comment:
I take it that it's ok to hold the lock during this period because nothing else should be legally happening at the class Server anyway, but could you add a comment to that effect? Otherwise it seems funny to have a lock around a timed while loop.